cranelift/codegen/src/isa/aarch64/abi.rs

```diff
@@ -238,6 +294,17 @@ impl AArch64ABIBody {
             stackslots.push(off);
         }
 
+        let stack_limit = match f
```

Comment:
Actually, one more thought: after reading through the rest of the patch, it seems that `stack_limit` is not used beyond the prologue. Given this, I think it would be fine to use `spilltmp` within `gen_prologue` itself, and then carefully validate that we don't keep it live across anything else that could implicitly use it. This includes any virtual registers at all: the register allocator is liable to pop in and do a surprise spill with a big offset somewhere, wiping out `spilltmp`.